addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -865,29 +869,27 @@ def unpickable_item(self,
                 location=location,
             )
 
-            if picking.exists() \
-                    and picking.state == 'assigned' \
-                    and original_picking_id is not None\
-                    and not picking.picking_type_id.u_post_assign_action:
-                # A backorder has been created, but the stock is
-                # available; get rid of the backorder after linking the
-                # move lines to the original picking, so it can be
-                # directly processed
-                picking.move_line_ids.write({'picking_id': original_picking_id})
-                picking.move_lines.write({'picking_id': original_picking_id})
-                picking.unlink()
-            else:
-                # Moves may be part of a new picking after refactor, this should
-                # be added back to the batch
-                moves.mapped('picking_id')\
-                    .filtered(lambda p: p.state == 'assigned')\
-                    .write({'batch_id': self.id})
-            self._remove_unready_picks()
-            self._compute_state()
-
-        else:
-            picking.batch_id = False
-
+            for picking in to_investigate:
+                moves = picking.move_lines
+                if picking.exists() \
+                        and picking.state == 'assigned' \
+                        and original_picking_ids[picking] is not None\
+                        and not picking.picking_type_id.u_post_assign_action:
+                    # A backorder has been created, but the stock is
+                    # available; get rid of the backorder after linking the
+                    # move lines to the original picking, so it can be
+                    # directly processed
+                    picking.move_line_ids.write({'picking_id': original_picking_id})
```

Comment:
This is still using the old `original_picking_id`.
We can just create a var before line 874 `original_picking_id = original_picking_ids[picking]` and use it also in line 876.